File: v131016/TeeNew/Chart_Grid.pas
unit Chart_Grid;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  QGrids,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  Grids,
  {$ENDIF}
  Base, TeeChartGrid, TeeProcs, TeEngine, Chart, Series, TeCanvas,
  TeeNavigator, TeePenDlg;

type
  TChartGridForm = class(TBaseForm)
    Series1: TBarSeries;
    Panel2: TPanel;
    ChartGridNavigator1: TChartGridNavigator;
    ChartGrid1: TChartGrid;
    CheckBox1: TCheckBox;
    CheckBox2: TCheckBox;
    ButtonColor1: TButtonColor;
    Splitter1: TSplitter;
    procedure FormCreate(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure CheckBox2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TChartGridForm.FormCreate(Sender: TObject);
begin
  inherited;
  ButtonColor1.LinkProperty(Series1,'SeriesColor');

  { fill series }
  Series1.Add( 32,'John');
  Series1.Add(417,'Anne');
  Series1.Add( 65,'Louise');
  Series1.Add( 87,'Jeff');

  { after adding points, we should notify the grid... }
  ChartGrid1.RecalcDimensions;
end;

procedure TChartGridForm.CheckBox1Click(Sender: TObject);
begin
  ChartGrid1.ShowLabels:=CheckBox1.Checked;
end;

procedure TChartGridForm.CheckBox2Click(Sender: TObject);
begin
  if CheckBox2.Checked then
     ChartGrid1.Options:=ChartGrid1.Options-[goEditing]
  else
     ChartGrid1.Options:=ChartGrid1.Options+[goEditing];

  { repaint navigator, disabling/enabling buttons }
  ChartGridNavigator1.EnableButtons;
end;

initialization
  RegisterClass(TChartGridForm);
end.

File: TeeNew/Builder2009/Chart_Grid.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Chart_Grid.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeChartGrid"
#pragma link "TeeNavigator"
#pragma resource "*.dfm"
TChartGridForm *ChartGridForm;
//---------------------------------------------------------------------------
__fastcall TChartGridForm::TChartGridForm(TComponent* Owner)
        : TBaseForm(Owner)
{

}
//---------------------------------------------------------------------------
void __fastcall TChartGridForm::FormCreate(TObject *Sender)
{
  ButtonColor1->LinkProperty(Series1,"SeriesColor");
  // fill series
  Series1->Add( 32,"John",clTeeColor);
  Series1->Add(417,"Anne",clTeeColor);
  Series1->Add( 65,"Louise",clTeeColor);
  Series1->Add( 87,"Jeff",clTeeColor);

  // after adding points, we should notify the grid...
  ChartGrid1->RecalcDimensions();
}
//---------------------------------------------------------------------------

void __fastcall TChartGridForm::CheckBox1Click(TObject *Sender)
{
  ChartGrid1->ShowLabels = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TChartGridForm::CheckBox2Click(TObject *Sender)
{
  if (CheckBox2->Checked) ChartGrid1->Options = ChartGrid1->Options - TGridOptions()<<goEditing;
  else ChartGrid1->Options = ChartGrid1->Options + TGridOptions()<<goEditing;

  // repaint navigator, disabling/enabling buttons
  ChartGridNavigator1->EnableButtons();
}
//---------------------------------------------------------------------------

